crates/forge/tests/integration/test_state.rs

```diff
@@ -331,3 +331,105 @@ fn test_cant_call_test_contract() {
     assert_failed!(result);
     assert_case_output_contains!(result, "test_calling_test_fails", "not deployed");
 }
+
+#[test]
+fn test_simple_cheatcodes() {
+    let test = test_case!(indoc!(
+        r#"
+        use result::ResultTrait;
+        use box::BoxTrait;
+        use serde::Serde;
+        use starknet::ContractAddress;
+        use array::SpanTrait;
+        use starknet::ContractAddressIntoFelt252;
+        use snforge_std::{
+            start_prank, stop_prank,
+            start_roll, stop_roll,
+            start_warp, stop_warp,
+            test_address
+        };
+
+        #[test]
+        fn test_prank_test_state() {
+            let test_address: ContractAddress = test_address();
+            let caller_addr_before = starknet::get_caller_address();
+            let target_caller_address: ContractAddress = (123_felt252).try_into().unwrap();
+
+            start_prank(test_address, target_caller_address);
+            let caller_addr_after = starknet::get_caller_address();
+            assert(caller_addr_after==target_caller_address, caller_addr_after.into());
+
+            stop_prank(test_address);
+            let caller_addr_after = starknet::get_caller_address();
+            assert(caller_addr_after==caller_addr_before, caller_addr_before.into());
+        }
+
+        #[test]
+        fn test_roll_test_state() {
+            let test_address: ContractAddress = test_address();
+            let old_block_number = starknet::get_block_info().unbox().block_number;
+
+            start_roll(test_address, 234);
+            let new_block_number = starknet::get_block_info().unbox().block_number;
+            assert(new_block_number == 234, 'Wrong block number');
+
+            stop_roll(test_address);
+            let new_block_number = starknet::get_block_info().unbox().block_number;
+            assert(new_block_number == old_block_number, 'Block num did not change back');
+        }
+
+        #[test]
+        fn test_warp_test_state() {
+            let test_address: ContractAddress = test_address();
+            let old_block_timestamp = starknet::get_block_info().unbox().block_timestamp;
+
+            start_warp(test_address, 123);
+            let new_block_timestamp = starknet::get_block_info().unbox().block_timestamp;
+            assert(new_block_timestamp == 123, 'Wrong block timestamp');
+
+            stop_warp(test_address);
+            let new_block_timestamp = starknet::get_block_info().unbox().block_timestamp;
+            assert(new_block_timestamp == old_block_timestamp, 'Timestamp did not change back')
+        }
+    "#
+    ));
+
+    let result = run_test_case(&test);
+
+    assert_passed!(result);
+}
+
+#[test]
+#[ignore] //TODO (#612)
```

Comment:
Either the description of this PR should be changed so #612 is not closed after merge of this, or we should create an separate issue for this test.

Reply:
Updated the description, so this PR will not close the issue